### ISSUE TITLE
[Feature Request] Support Activity Result APIs for AuthUI

### ISSUE THREAD

**Issue by jossiwolf** [firebase/FirebaseUI-Android#1916]
The Android Developers documentation [recommends using the Activity Result APIs](https://developer.android.com/training/basics/intents/result) instead of relying on intents for launching activities for results. Using the new APIs would enable a nicer development experience. Take the following as an example:

```kotlin
class FirebaseAuthUIActivityResult: ActivityResultContract<AuthUI.SignInIntentBuilder, IdpResponse>() { ... }
```

This API could be used like this:

```kotlin
val firebaseLogin = registerForActivityResult(FirebaseAuthUIActivityResult()) { result ->
    if (result.resultCode == Activity.RESULT_OK) {
        ...
    }
}
firebaseLogin.launch(
    AuthUI.getInstance().createSignInIntentBuilder()
        .setAvailableProviders(authenticationMethods)
        .setTheme(R.style.SignInTheme)
        .setAuthMethodPickerLayout(signInLayout)
)
```

By using a contract with types, this is safer for the end-user. More importantly, it enables future usage from Jetpack Compose.
I'd be happy to contribute here if we can scope it a bit :)

Let me know what you think!

**Comment by samtstern** [comment]
@jossiwolf this sounds like a great feature improvement and I would be happy to accept contributions to fix this. The first questions we need to answer are:

1. What dependencies do we need to add or change to make this possible?
2. What downstream effects do those dependencies have on FirebaseUI developers? Do they require using Java 8 or Kotlin or are they compatible with "traditional" Java Android apps?

The answers to those questions will determine if we can do this in a feature release like `7.2.0` or if we need to wait until the `8.0.0` breaking change.

**Comment by jossiwolf** [comment]
@samtstern Awesome! Thanks for looking at this so quickly :)

1. The Activity Result APIs live in the `androidx.activity:activity` (or the `ktx` version for Kotlin support) artefact. This should be pulled in as a dependency of AppCompat already, but we might need to declare a version constraint (we need version 1.2.0 of the `androidx.activity` artefact) or just explicitly depend on it
2. The Kotlin version of `androidx.activity` requires Java 8 language features, the default version shouldn't have any effect on downstream users. I'll have to look into it, but I think if we shipped this in Java and a user declared a dependency on the `androidx.activity:activity-ktx`  artefact they should still be able to use the nice Kotlin features on their end, so there's probably no need for us to include the `ktx` artefact.

It might make sense to introduce this API in a feature release before `8.0.0` and deprecate the old APIs with that move and remove them with `8.0.0`.

**Comment by samtstern** [comment]
@jossiwolf that sounds good to me. If you want to try and prepare a PR targeting the `version-7.2.0-dev` branch I am happy to review it and work with you to get it merged.

**Comment by samtstern** [comment]
This was fixed and released in `7.2.0`, thank you @jossiwolf !
